api/src/main/java/eu/coatrack/api/Proxy.java

```diff
@@ -157,6 +164,30 @@ public void setPort(Integer port) {
         this.port = port;
     }
 
+    public Long getLastCallTimeToAdmin() {
```

Comment:
If a variable has getters and setters, it is not encapsulated anymore and could maybe be set to public. This could remove verbose code.

Reply:
So do you mean that the "public" should be changed?

Reply:
What I meant was, that a private variable with a getter and a setter does exactly the same as a public variable. My suggestion was to remove getters/setters and make the variable 'lastCallTimeToAdmin' itself public instead of private which would make the code less verbose. This strategy could be considered for other variables as well.

Reply:
Yes, I understand now what you mean.
However for different purposes and advantages it is best pratice to use getters and setters instead of a public attribute. Please check https://stackoverflow.com/questions/11071407/advantage-of-set-and-get-methods-vs-public-variable
https://stackoverflow.com/questions/1568091/why-use-getters-and-setters-accessors

Reply:
These are good arguments for your position and now I agree with you. However, I noticed an other problem: 

The setter of this variable is used only once: ```proxy.setTimeOfLastSuccessfulCallToAdmin(LocalDateTime.now());```. The purpose is to update ```timeOfLastSuccessfulCallToAdmin``` but the logic (= call of "now()") is used outside the class. 

Encapsulation should not be weakened by necessary setters and tightly coupled data and behavior/logic should be within the same class. A good example of what I mean can be found here: https://martinfowler.com/bliki/TellDontAsk.html

My suggestion is to delete the setter method and add an ```updateTimeOfLastSuccessfulCallToAdmin``` to your class which sets the ```timeOfLastSuccessfulCallToAdmin``` to now.

Reply:
I found the TellDontAsk very interesting.
So I agree with you and applied that idea.
Done